WikiExtension/src/timeSeriesService.js

```diff
@@ -25,24 +26,97 @@ const fetchPageViews = async (title, startDate, endDate, aggregateType) => {
     return await response.json();
 };
 
+/**
+ * Get all the revision objects on a given wikipedia title.
+ *
+ * The api only gives us at most 20 reivision objects per request. However, it does give us a query to
+ * use if there are any older revisions. If older versions of revision exists, we will get
+ * {
+ *  revisions: [...],
+ *  ...
+ *  older: 'https://en.wikipedia.org/w/rest.php/v1/page/Jupiter/history?older_than=1103899458',
+ *  ...
+ * }
+ *
+ * Thus, we keep calling the endpoint with the given query if the older property exists in the
+ * previous request.
+ *
+ * @param {string} title of a wikipedia article
+ * @param {Date} startDate
+ * @param {Date} endDate
+ * @returns An array containing all revision objects
+ */
+const fetchPageRevisions = async (title, startDate, endDate) => {
+    const response = await fetch(`https://en.wikipedia.org/w/rest.php/v1/page/${title}/history`);
+    let curJsonResponse = await response.json();
+
+    let revisions = curJsonResponse.revisions;
+
+    while (curJsonResponse.hasOwnProperty("older")) {
+        const localResponse = await fetch(curJsonResponse.older);
+        curJsonResponse = await localResponse.json();
+
+        const localRevisions = curJsonResponse.revisions;
+
+        // Stop making requests when requesting passes startDate
+        const localLatestTimestamp = new Date(localRevisions[0].timestamp);
+        if (localLatestTimestamp < startDate) break;
+
+        revisions.push(...localRevisions);
```

Comment:
Is this a typo? the "..."

Reply:
nope. it is for adding all the elements in the list into a current list. If we use `append`, we will have to write it as something like 
```
let revisions = ...
...
revisions = revisions.append(localRevisions)
```
which I think is less clean.

Reply:
Nice, that's great, didn't know they had this functionality